### ISSUE TITLE
Null check operator on null value if search for a NodeValue without set Type

### ISSUE THREAD

**Issue by nicolaeschbach** [cyber-geiger/toolbox-storage#35]
**Describe the bug**
If a searchCriteria is set up to listen to a change on a value in a given path a exception can occur if the Field.Type of the found value is null.

**Environmental information**
geiger_localstorage: ^0.6.21

**To Reproduce**
void main() async {
  test('Register NodeValue Change Listener', () async {
    StorageController controller = GenericController("testOwner", DummyMapper("testOwner"));
    SearchCriteria criteria = SearchCriteria(searchPath: ":Local", key: "currentDevice");
    controller.registerChangeListener(UpdateUUIDListener(), criteria);
    controller.updateValue(":Local", NodeValueImpl("currentDevice", "newDeviceUUID"));
  });
}

class UpdateUUIDListener implements StorageListener {

  // Update on currentDeviceChange the networkInterface uuid
  @override
  void gotStorageChange(EventType event, Node? oldNode, Node? newNode) async {
    String? value = (await newNode!.getValue("currentDevice"))!.value;
    print (value);
  }
}

**Expected behavior**
The Code above should run without Exception and notify changes and print the new value to console

**Comment by martingwerder** [comment]
I have added your testcase to the user tests. Unfortunately I cannot reproduce the behavior above. Either it was fixed "accidentally" or there is something wrong with the testcase. Can you confirm or deny that the issue still exists with 0.6.22?

**Comment by nicolaeschbach** [comment]
This is still an issue with 0.6.22. 
Note: for the LocalStorage Replikation it could be handled in another way. But it seems for me to be a bug that should be fixed anyway.

This is stack-trace of the error:
```
C:\tools\flutter\bin\flutter.bat --no-color test --machine --start-paused --plain-name "Register NodeValue Change Listener" --concurrency=1 test\github_issue_test.dart
Testing started at 13:23 ...

package:geiger_localstorage/src/search_criteria.dart 196:49        SearchCriteria.evaluate
===== asynchronous gap ===========================
dart:async                                                         _asyncErrorWrapperHelper
package:geiger_localstorage/src/db/generic_controller.dart 324:5   GenericController._addOrUpdateValue
===== asynchronous gap ===========================
dart:async                                                         _asyncThenWrapperHelper
package:geiger_localstorage/src/db/generic_controller.dart 95:15   GenericController._initMapper
===== asynchronous gap ===========================
dart:async                                                         _asyncThenWrapperHelper
package:geiger_localstorage/src/db/generic_controller.dart 330:11  GenericController.updateValue
test\github_issue_test.dart 76:22                                  main.<fn>
===== asynchronous gap ===========================
dart:async                                                         _asyncThenWrapperHelper
test\github_issue_test.dart                                        main.<fn>
Null check operator used on a null value
package:geiger_localstorage/src/search_criteria.dart 196:49        SearchCriteria.evaluate
===== asynchronous gap ===========================
dart:async                                                         _asyncErrorWrapperHelper
package:geiger_localstorage/src/db/generic_controller.dart 344:5   GenericController._updateValue
===== asynchronous gap ===========================
dart:async                                                         _asyncThenWrapperHelper
package:geiger_localstorage/src/db/generic_controller.dart 331:18  GenericController.updateValue
===== asynchronous gap ===========================
dart:async                                                         _asyncThenWrapperHelper
test\github_issue_test.dart 76:22                                  main.<fn>
===== asynchronous gap ===========================
dart:async                                                         _asyncThenWrapperHelper
test\github_issue_test.dart                                        main.<fn>

Null check operator used on a null value


```

**Comment by martingwerder** [comment]
OK... that was an easy straight forward one. I really can't tell why I missed it. Sorry. I released it as bugfix release in 0..23. Can you confirm the fix?

**Comment by nicolaeschbach** [comment]
Is fixed with 0.6.23
